bin/samlocal

```diff
@@ -78,7 +81,25 @@ def is_ecr_url(url: str) -> bool:
 is_ecr_url_orig = ecr_utils.is_ecr_url
 ecr_utils.is_ecr_url = is_ecr_url
 types.is_ecr_url = is_ecr_url
-
+types.ImageRepositoryType.transformer = types.ImageRepositoryType.Transformer(
+    converter=click.STRING,
+    transformation=is_ecr_url
+)
+
+def upload(self, image, resource_name):
+    try:
+        ecr_endpoint = get_registry_url(self)
+        if "amazonaws.com" in self.ecr_repo_multi.get(resource_name):
+            self.ecr_repo_multi[resource_name] = self.ecr_repo_multi[resource_name].replace("amazonaws.com", ecr_endpoint)
+    except (AttributeError, TypeError):
+        pass
+    return upload_orig(self, image,resource_name)
+
+def get_registry_url(self):
+    return self.ecr_client.describe_repositories()["repositories"][0]["repositoryUri"].split("/",1)[0]
```

Comment:
Can we always assume that we want to use the first repo at index `0` here?

Reply:
Yes, as we only want to get the Localstack endpoint for ecr from boto so we can override with it the upload function.
The API response for describe repositories is something like this:
```json
{
    "repositories": [
        {
...
    "repositoryUri": "000000000000.dkr.ecr.us-east-1.localhost.localstack.cloud:4510/statement-process-images",
...
        }
    ]
}
```
So we can safely assume unless boto is returning with the repos of multiple regions we always end up with the right value.